surefire-its/src/test/resources/surefire-1967-testng-method-parallel-ordering/pom.xml

```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.apache.maven.surefire</groupId>
+    <artifactId>it-parent</artifactId>
+    <version>1.0</version>
+    <relativePath>../pom.xml</relativePath>
+  </parent>
+
+  <groupId>org.apache.maven.plugins.surefire</groupId>
+  <artifactId>surefire-1967-testng-method-parallel-ordering</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <name>TestNG parallel ordering</name>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.testng</groupId>
+      <artifactId>testng</artifactId>
+      <version>${testNgVersion}</version>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>${surefire.version}</version>
+        <configuration>
```

Comment:
This is rather a cosmetic change proposal

`Base.java`

```
        int mavenSurefireThreadCount = Integer.parseInt(System.getProperty( "surefire.thread-count" ));
        int concurrentResources = resources.incrementAndGet();
        if (concurrentResources > mavenSurefireThreadCount) {
            throw new IllegalStateException(String.format("Tests execute in two threads, so there should be at most %d resources allocated, got: %d", mavenSurefireThreadCount, concurrentResources));
        }
```


`pom.xml`

```
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-surefire-plugin</artifactId>
        <version>${surefire.version}</version>
        <configuration>
          <threadCount>2</threadCount>
          <parallel>methods</parallel>
          <systemPropertyVariables>
              <surefire.thread-count>2</surefire.thread-count>
          </systemPropertyVariables>
        </configuration>
```
Obviously the `2` value in `pom.xml` can be passed along (same as `${surefire.version}`) to avoid duplication.

The main advantage in adding this cosmetic change is that you could easily test the behaviour with multiple values for the `threadCount` (`1` , `2`, `3` and a value greater than the number of methods being tested).

Reply:
> The main advantage in adding this cosmetic change is that you could easily test the behaviour with multiple values for the `threadCount`

i don't see advantage of doing this. Please help me understand.

Reply:
You currently hardcoded the logic for checking the concurrency level in the `Base` class to be `2`.
By using a `sysProp` for the thread count in `Surefire1967CheckTestNgMethodParallelOrderingIT` you could test the functionality not only with different `TestNG` versions, but also with different `threadCount` values (serial - `1` , concurrent - `2` , and also a greater value than the number of methods in the test suite).

Reply:
I understand that parameterizing threadCount would allow me to run tests with different threadCount values. Would we actually want to run the test more times? What additional value would it bring?

Reply:
I see an advantage in seeing whether we test multiple values (at least `1` and `2` ) for the thread count.

Looking closer at `Base` class we could add a `@Before` setup and `@After`  and tear down methods to verify whether the amount of tests being executed in parallel is indeed between `1` and the configured `threadCount`.